services/dbaas/cluster.go

```diff
@@ -825,6 +990,102 @@ func SetPgSqlClusterData(d *schema.ResourceData, cluster psql.ClusterResponse) e
+
+	if cluster.Id != nil {
+		d.SetId(*cluster.Id)
+	}
+
+	if cluster.Properties == nil {
+		return fmt.Errorf("response properties should not be empty for MariaDB cluster with ID %v", *cluster.Id)
+	}
+
+	if cluster.Properties.MariadbVersion != nil {
+		if err := d.Set("mariadb_version", *cluster.Properties.MariadbVersion); err != nil {
+			return utils.GenerateSetError(resourceName, "mariadb_version", err)
+		}
+	}
+
+	if cluster.Properties.Instances != nil {
+		if err := d.Set("instances", *cluster.Properties.Instances); err != nil {
+			return utils.GenerateSetError(resourceName, "instances", err)
+		}
+	}
+
+	if cluster.Properties.Cores != nil {
+		if err := d.Set("cores", *cluster.Properties.Cores); err != nil {
+			return utils.GenerateSetError(resourceName, "cores", err)
+		}
+	}
+
+	if cluster.Properties.Ram != nil {
+		if err := d.Set("ram", *cluster.Properties.Ram); err != nil {
+			return utils.GenerateSetError(resourceName, "ram", err)
+		}
+	}
+
+	if cluster.Properties.StorageSize != nil {
+		if err := d.Set("storage_size", *cluster.Properties.StorageSize); err != nil {
+			return utils.GenerateSetError(resourceName, "storage_size", err)
+		}
+	}
+
+	if cluster.Properties.Connections != nil && len(*cluster.Properties.Connections) > 0 {
+		var connections []interface{}
+		for _, connection := range *cluster.Properties.Connections {
+			connectionEntry := c.SetConnectionProperties(connection)
+			connections = append(connections, connectionEntry)
+		}
+		if err := d.Set("connections", connections); err != nil {
+			return utils.GenerateSetError(resourceName, "connections", err)
+		}
+	}
+
+	if cluster.Properties.DisplayName != nil {
+		if err := d.Set("display_name", *cluster.Properties.DisplayName); err != nil {
+			return utils.GenerateSetError(resourceName, "display_name", err)
+		}
+	}
+
+	if cluster.Properties.MaintenanceWindow != nil {
+		var maintenanceWindow []interface{}
+		maintenanceWindowEntry := c.SetMaintenanceWindowProperties(*cluster.Properties.MaintenanceWindow)
+		maintenanceWindow = append(maintenanceWindow, maintenanceWindowEntry)
+		if err := d.Set("maintenance_window", maintenanceWindow); err != nil {
+			return utils.GenerateSetError(resourceName, "maintenance_window", err)
+		}
+	}
+
+	if cluster.Properties.DnsName != nil {
+		if err := d.Set("dns_name", *cluster.Properties.DnsName); err != nil {
+			return utils.GenerateSetError(resourceName, "dns_name", err)
+		}
+	}
+
+	return nil
+}
+
+func (c *MariaDBClient) SetConnectionProperties(connection mariadb.Connection) map[string]interface{} {
+	connectionMap := map[string]interface{}{}
+
+	utils.SetPropWithNilCheck(connectionMap, "datacenter_id", connection.DatacenterId)
+	utils.SetPropWithNilCheck(connectionMap, "lan_id", connection.LanId)
+	utils.SetPropWithNilCheck(connectionMap, "cidr", connection.Cidr)
+
+	return connectionMap
+}
+
+func (c *MariaDBClient) SetMaintenanceWindowProperties(maintenanceWindow mariadb.MaintenanceWindow) map[string]interface{} {
```

Comment:
we should be able to use an interface here based on the getters. For another time.